pydash/predicates.py

```diff
@@ -535,6 +537,9 @@ def is_equal(value, other, callback=None):
     return equal
 
 
+is_equal_with = is_equal
```

Comment:
`is_equal_with` and `is_equal` should be separate functions. Lodash split the original functionality of `is_equal` into `is_equal` and `is_equal_with`. Basically, `is_equal_with` should work just like `is_equal` does previously where it accepts a callback. Then `is_equal` should become a new function that doesn't allow a callback and instead just calls `is_equal_with`.